axiom/nr_distributed_trace.c

```diff
@@ -1298,3 +1298,42 @@ char* nr_distributed_trace_create_w3c_traceparent_header(const char* trace_id,
 
   return trace_parent_header;
 }
+
+void nr_distributed_trace_handle_inbound_w3c_sampled_flag(
+        nr_distributed_trace_t* dt,
+        const nrobj_t* trace_headers,
+        int remote_parent_sampled,
+        int remote_parent_not_sampled) {
+  const nrobj_t* traceparent = NULL;
+  int sampled = 0;
+  nr_status_t* parse_err = 0;
+  if (0 != remote_parent_sampled || 0 != remote_parent_not_sampled) {
+    traceparent = nro_get_hash_value(trace_headers, "traceparent", parse_err);
+    if (nrunlikely(NULL == traceparent || NR_SUCCESS != parse_err)) {
+      return;
+    }
+    sampled = nro_get_hash_int(traceparent, "trace_flags", parse_err);
```

Comment:
According to [Trace Context HTTP Request Headers Format specification](https://www.w3.org/TR/trace-context-2/), `trace-flags` is hex-encoded. Can `nro_get_hash_int` handle that?

Reply:
At this point, the traceparent string has already been deconstructed with regex here: https://github.com/newrelic/newrelic-php-agent/blob/f095db28779d0dd01429542a708322e423ef22aa/axiom/nr_distributed_trace.c#L853

The trace_flags hex string is converted from hex with `strtol(_, _, 16);`

Reply:
This will work because when `traceparent` header is converted to `nro`, `trace-flags` are treated as hex-encoded field - see https://github.com/newrelic/newrelic-php-agent/blob/f095db28779d0dd01429542a708322e423ef22aa/axiom/nr_distributed_trace.c#L935-L937